nixos/modules/hardware/temp-throttle.nix

```diff
@@ -0,0 +1,64 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.temp-throttle;
+  configFile = with builtins;
+    "MAX_TEMP=" + toString cfg.max_temp + "\n" +
+    (if cfg.interval  != null then "INTERVAL="  + toString cfg.interval + "\n" else "") +
+    (if cfg.temp_file != null then "TEMP_FILE=" + cfg.interval          + "\n" else "") +
+    (if cfg.core      != null then "CORE="      + toString cfg.core     + "\n" else "") +
+    (if cfg.log_file  != null then "LOG_FILE="  + cfg.interval          + "\n" else "")
```

Comment:
1. Use string interpolation, e.g. `MAX_TEMP=${cfg.max_temp}`.
2. Use `lib.optionalString` instead of `if … then … else ""`.